src/screens/signup/SignUp.js

```diff
@@ -0,0 +1,292 @@
+import styled from "styled-components/native";
+import { flexRowMarginXView, longKiwiButton } from "../../styles/mixin";
+import {
+  SIGN_UP_API,
+  VALIDATE_NICKNAME_API,
+  KIWI_REST_API_KEY,
+} from "../../config";
+import { AsyncStorage } from "react-native";
+
+const MAX_NICKNAME_LENGTH = 20;
+
+const SignUp = ({ navigation, route }) => {
+  const { phoneNumber = "" } = route.params;
+  const [nickname, setNickname] = useState("");
+  const [focusNickname, setFocusNickname] = useState(false);
+  const [checkNickname, setCheckNickname] = useState(false);
+  const [isValidNickname, setIsValidNickname] = useState(false);
+  const [myTown, setMyTown] = useState("동내를 인증해주세요!");
+  const [townCode, setTownCode] = useState("");
+  const [coordinate, setCoordinate] = useState();
+
+  const goBack = () => navigation.pop();
+
+  const getCurrentLocation = async () => {
+    setMyTown("동내 인증중...");
+    const { status } = await Location.requestPermissionsAsync();
+    if (status !== "granted") {
+      Alert("현재 위치에 대한 정보를 받아 오고싶키위🥝");
+      return;
+    }
+    const myLocation = await Location.getCurrentPositionAsync({});
+    setCoordinate({
+      latitude: myLocation.coords.latitude,
+      longitude: myLocation.coords.longitude,
+    });
+    const townName = await getTownName([
+      myLocation.coords.latitude,
+      myLocation.coords.longitude,
+    ]);
+
+    setMyTown(townName);
+  };
+
+  const getTownName = async (coordinate) => {
+    const response = await fetch(
+      `https://dapi.kakao.com/v2/local/geo/coord2regioncode.json?x=${coordinate[1]}&y=${coordinate[0]}`,
+      {
+        method: "GET",
+        headers: {
+          Authorization: `KakaoAK ${KIWI_REST_API_KEY}`,
+        },
+      }
+    );
+    const result = await response.json();
+
+    // [0]: 법정동, [1]: 행정동
+    setTownCode(result.documents[0].code);
+    return result.documents[0].address_name;
+  };
+
+  const resetNicknameState = () => {
+    setNickname("");
+    setCheckNickname(false);
+    setIsValidNickname(false);
+  };
+
+  const TrySignUp = async () => {
+    try {
+      const response = await fetch(SIGN_UP_API, {
+        method: "POST",
+        body: JSON.stringify({
+          phone_number: phoneNumber,
+          nickname: nickname,
+          address_code: townCode,
+          email: "",
+        }),
+      });
+      const result = await response.json();
+      if (result.message === "SUCCESS") {
+        AsyncStorage.setItem("token", result.token);
+        navigation.replace("BottomTabNavigator");
+        alert("가입해 주셔서 감사합니다!");
+      } else if (result.message === "DUPLICATED_NICKNAME") {
+        alert("중복된 닉네임 입니다");
+        resetNicknameState();
+      } else {
+        alert(result.message);
+      }
+    } catch {}
+  };
+
+  const validateNickname = async () => {
+    try {
+      const response = await fetch(VALIDATE_NICKNAME_API, {
+        method: "POST",
+        body: JSON.stringify({
+          nickname: nickname,
```

Comment:
key, value가 같으면 줄일 수 있습니다!
```suggestion
          nickname,
```

Reply:
앗 관희님 투두리스트 보고 알게된 문법이었지만, 역시 써보기 전까지는.... ㅋㅋㅋㅋㅋㅋ